src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar.kt

```diff
@@ -169,6 +172,29 @@ public abstract class ShadowJar : Jar() {
   @get:Option(option = "relocation-prefix", description = "Prefix used for auto relocation of packages in the dependencies.")
   public open val relocationPrefix: Property<String> = objectFactory.property(ShadowBasePlugin.SHADOW)
 
+  /**
+   * The [org.gradle.api.attributes.Bundling] attribute to use for the Gradle Module Metadata.
+   *
+   * If unset, Shadow will use [org.gradle.api.attributes.Bundling.SHADOWED] if this shadow JAR is relocating packages
+   * or [org.gradle.api.attributes.Bundling.EMBEDDED] if it is not.
+   *
+   * Defaults to `shadow`.
+   */
+  @get:Internal
+  public open val bundlingAttribute: Property<String> = objectFactory.property(Bundling.SHADOWED)
+
+  @Internal
+  internal fun getFinalBundlingAttribute(): Provider<Bundling> {
+    return bundlingAttribute.orElse(
+      providers.zip<Set<Relocator>, Boolean, Boolean>(
+        relocators,
+        enableAutoRelocation,
+      ) { set, autoRelocating ->
+        !set.isEmpty() || autoRelocating
+      }.orElse(false).map { isRelocating -> if (isRelocating) Bundling.SHADOWED else Bundling.EMBEDDED },
+    ).map { objectFactory.named(Bundling::class.java, it) }
+  }
```

Comment:
We don't need this `getFinalBundlingAttribute`, it's a behavior change. If users want to override the default bundling property, just set `bundlingAttribute`.